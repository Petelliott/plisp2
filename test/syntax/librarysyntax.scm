(println (let ((a (+ 1 2))
               (b 5))
           (newline)
           (+ a b)))

(println (let* ((a 5)
                (b (+ 5 4)))
           (newline)
           (+ b 1)))

(println (begin) (begin 5) (begin (write 'a) 6))

(println (cond)
         (cond
          (#f 1)
          (#f 2))
         (cond
          (#f (newline))
          (#t 4)
          (#f (newline)))
         (cond
          (#f 5)
          (else 6)))

(println (and)
         (and 1 2)
         (and 1 #f)
         (and 1 #f (newline) (newline)))

(println (or)
         (or 1 2)
         (or 1 (newline) (newline))
         (or #f #f 2))

(println (when #f (write 'a) (write 'b) 8)
         (when #t 7))

(println (unless #t (write 'a) (write 'b) 8)
         (unless #f 7))
