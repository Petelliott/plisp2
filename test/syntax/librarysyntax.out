
8

10
a#<unspecified> 5 6
#<unspecified> #<unspecified> 4 6
#t 2 #f #f
#f 1 1 2
#<unspecified> 7
#<unspecified> 7
